ml-agents/mlagents/trainers/components/reward_signals/gail/model.py

```diff
@@ -0,0 +1,261 @@
+                    self.policy_model.normalize_vector_obs(self.policy_model.vector_in)
+                )
+            else:
+                encoded_expert_list.append(self.obs_in_expert)
+                encoded_policy_list.append(self.policy_model.vector_in)
+
+        if self.policy_model.vis_obs_size > 0:
+            self.expert_visual_in: List[tf.Tensor] = []
+            visual_policy_encoders = []
+            visual_expert_encoders = []
+            for i in range(self.policy_model.vis_obs_size):
+                # Create input ops for next (t+1) visual observations.
+                visual_input = self.policy_model.create_visual_input(
+                    self.policy_model.brain.camera_resolutions[i],
+                    name="visual_observation_" + str(i),
+                )
+                self.expert_visual_in.append(visual_input)
+
+                encoded_policy_visual = self.policy_model.create_visual_observation_encoder(
+                    self.policy_model.visual_in[i],
+                    self.encoding_size,
+                    LearningModel.swish,
+                    1,
+                    "stream_{}_visual_obs_encoder".format(i),
+                    False,
+                )
+
+                encoded_expert_visual = self.policy_model.create_visual_observation_encoder(
+                    self.expert_visual_in[i],
+                    self.encoding_size,
+                    LearningModel.swish,
+                    1,
+                    "stream_{}_visual_obs_encoder".format(i),
+                    True,
+                )
+                visual_policy_encoders.append(encoded_policy_visual)
+                visual_expert_encoders.append(encoded_expert_visual)
+            hidden_policy_visual = tf.concat(visual_policy_encoders, axis=1)
+            hidden_expert_visual = tf.concat(visual_expert_encoders, axis=1)
+            encoded_policy_list.append(hidden_policy_visual)
+            encoded_expert_list.append(hidden_expert_visual)
+
+        self.encoded_expert = tf.concat(encoded_expert_list, axis=1)
+        self.encoded_policy = tf.concat(encoded_policy_list, axis=1)
+
+    def create_encoder(
+        self, state_in: tf.Tensor, action_in: tf.Tensor, done_in: tf.Tensor, reuse: bool
+    ) -> Tuple[tf.Tensor, tf.Tensor]:
+        """
+        Creates the encoder for the discriminator
+        :param state_in: The encoded observation input
+        :param action_in: The action input
+        :param done_in: The done flags input
+        :param reuse: If true, the weights will be shared with the previous encoder created
+        """
+        with tf.variable_scope("GAIL_model"):
+            if self.use_actions:
+                concat_input = tf.concat([state_in, action_in, done_in], axis=1)
+            else:
+                concat_input = state_in
+
+            hidden_1 = tf.layers.dense(
+                concat_input,
+                self.h_size,
+                activation=tf.nn.elu,
```

Comment:
Do we want to use swish for these activations function as well (for consistency) ?

Reply:
Yes. Related note: should we call it `LearningModel.activation` rather than `LearningModel.swish` so that if we change it in the future we can just change one spot?

Reply:
I think that makes sense.

Reply:
Will leave that for a future PR, for now changed to `swish`